src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -31,36 +31,32 @@ public void put(K key, V value) {
         if (size > threshold) {
             resize();
         }
-
-        int index = hashKey(key);
+        int index = getIndex(key);
         Node<K, V> addNode = table[index];
-        if (addNode == null) {
-            table[index] = new Node<K,V>(key,value,null);
-            size++;
-            return;
-        }
-
         while (addNode != null) {
             if (addNode.key == key || addNode.key != null && addNode.key.equals(key)) {
                 addNode.value = value;
                 return;
             }
             if (addNode.nextNode == null) {
-                addNode.nextNode = new Node<K,V>(key,value,null);
+                addNode.nextNode = new Node<>(key,value,null);
                 size++;
                 return;
             }
             addNode = addNode.nextNode;
         }
+        table[index] = new Node<>(key,value,null);
+        size++;
+        return;
```

Comment:
do we need `return` at the end of the method?

Reply:
no, done